packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,202 @@
+   * - Optionally provide an `onClick` callback, which is fired when the link is clicked (pressed and released) with a mouse or keyboard. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/web/utilities/onlinknavigationprovider) to learn more about link navigation.
+   */
+  link?: {|
+    accessibilityLabel?: string,
+    externalLinkIcon?: ExternalLinkIcon,
+    href: string,
+    onClick?: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+    ref?: Ref<'a'>,
+    text: string,
+    target?: null | 'self' | 'blank',
+  |},
+  /**
+   * Callback fired when HelpIcon is clicked (pressed and released) with a mouse or keyboard.
+   */
+  onClick?: OnTapType,
+  /**
+   * Informational content that's displayed when the user clicks on HelpButton.
+   */
+  text: string | Element<typeof Text>,
+  /**
+   * Specifies the z-index for HelpButton's tooltip and popover to resolve any layering issues with other elements. See the [zIndex variant](https://gestalt.pinterest.systems/web/helpbutton#With-Z-index) for more details.
+   */
+  zIndex?: Indexable,
+|};
+
+/**
+ * [HelpButton](https://gestalt.pinterest.systems/web/helpbutton) is an affordance that accompanies an element on the screen. It helps describe or provide assistance on how to use the accompanying element.
+ */
+export default function HelpButton({
+  accessibilityPopoverLabel,
+  idealDirection = 'down',
+  link,
+  onClick,
+  text,
+  zIndex,
+}: Props): Node {
+  const tapAreaRef = useRef(null);
+  const [open, setOpen] = useState(false);
+  const [hovered, setHovered] = useState(false);
+  const [focused, setFocused] = useState(false);
+  const { name: colorSchemeName } = useColorScheme();
+  const popoverId = useId();
+  const { accessibilityTooltipMessage, accessibilityIconLabel } =
+    useDefaultLabelContext('HelpButton');
+
+  useEffect(() => {
+    if (open) {
+      tapAreaRef?.current?.focus();
+    }
+  }, [open]);
+
+  const toggleView = () => {
+    setOpen((currVal) => !currVal);
+  };
+
+  const onHandleTap = (props) => {
+    toggleView();
+    onClick?.(props);
+  };
+
+  const bgIconColor = open || hovered || focused ? 'default' : 'subtle';
+
+  const tooltipZIndex = zIndex ?? new FixedZIndex(1);
+
+  const zIndexWrapper = new CompositeZIndex([new FixedZIndex(tooltipZIndex.index())]);
```

Comment:
merged the CompositeZIndex Flow change. Now this can be refactored

Reply:
Done